lib/zip/entry_set.rb

```diff
@@ -13,8 +13,10 @@ def include?(entry)
       @entry_set.include?(to_key(entry))
     end
 
-    def find_entry(entry)
-      @entry_set[to_key(entry)]
+    def find_entry(entry, case_sensitively = true)
+      return @entry_set[to_key(entry)] if case_sensitively
+      entry = @entry_set.find { |k, _| k.downcase == to_key(entry).downcase }
```

Comment:
Guys, come on, this is suboptimal.

What should be done instead:
`
def to_key
   result = (blah blah ...)
   result = result.downcase! if options.case_insensitive_search
   result
end
`

This way, the entire code will operate properly in case if `options.case_insensitive_search` is `true`.